src/main/scala/is/hail/annotations/StagedRegionValueBuilder.scala

```diff
@@ -0,0 +1,174 @@
+package is.hail.annotations
+
+import is.hail.asm4s.{AsmFunction2, Code, FunctionBuilder}
+import is.hail.asm4s.Code._
+import is.hail.asm4s._
+import is.hail.expr._
+import is.hail.utils._
+import org.objectweb.asm.tree.{AbstractInsnNode, IincInsnNode}
+
+import scala.collection.generic.Growable
+import scala.reflect.ClassTag
+
+class StagedRegionValueBuilder[T] private (val fb: FunctionBuilder[AsmFunction2[T, MemoryBuffer, Long]], val rowType: Type, var region: Code[MemoryBuffer], val pOffset: Code[Long])(implicit tti: TypeInfo[T]) {
```

Comment:
I prefer the name `t` or `typ` instead of `rowType` because we can use an SRVB to produce a variety of types!